packages/plugin-core/src/commands/DeleteNodeCommand.ts

```diff
@@ -41,6 +49,89 @@ export class DeleteNodeCommand extends BasicCommand<
     return {};
   }
 
+  getBacklinkFrontmatterLineOffset(opts: {
+    link: DLink;
+    vaults: DVault[];
+    notes: NotePropsDict;
+    wsRoot: string;
+  }) {
+    const { link, vaults, notes, wsRoot } = opts;
+    const vault = VaultUtils.getVaultByName({
+      vaults,
+      vname: link.from.vaultName as string,
+    }) as DVault;
+    const noteWithLink = NoteUtils.getNoteByFnameV5({
+      fname: link.from.fname as string,
+      vault,
+      notes,
+      wsRoot,
+    }) as NoteProps;
+    const fsPath = NoteUtils.getFullPath({
+      note: noteWithLink,
+      wsRoot,
+    });
+    const fileContent = fs.readFileSync(fsPath).toString();
```

Comment:
this is not the greatest heuristic since we could have quoted --- inside the frontmatter. 
we have frontmatter logic in a few places now that parse using unified. can we use that and also consolidate?
- [moveCursorPastFrontmatter](https://github.com/dendronhq/dendron/blob/8fd685d53e8dd31f19bcb6e8d3b5ded25d86a63d/packages/plugin-core/src/windowWatcher.ts#L164)
- [provideFoldingRanges](https://github.com/dendronhq/dendron/blob/1f0b25a112e3da630d38cfcf7966003cd8520984/packages/plugin-core/src/features/FrontmatterFoldingRangeProvider.ts#L17)

can we consolidate the frontmatter detection logic to [RemarkUtils](https://github.com/dendronhq/dendron/blob/90e083b5e10073acbc8967ad9649c0008aae381c/packages/engine-server/src/markdown/remark/utils.ts#L872) ?